File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/2023s/IZ08IO2/BridgeProject/src/ShapeTriangle.java

import java.awt.Color;
import java.awt.Graphics;

public class ShapeTriangle extends ShapeBase
{
    private int ax,ay,bx,by,cx,cy;
    public ShapeTriangle(String text)
    {
        parse(text, (v)->ax=v, (v)->ay=v, (v)->bx=v, (v)->by=v, (v)->cx=v, (v)->cy=v);
    }
    
    @Override public void Draw(Graphics g)
    {
        g.setColor(Color.blue);
        int[] xs={ax,bx,cx,ax},ys={ay,by,cy,ay};
        g.drawPolyline(xs,ys,4);
    }
    
    @Override public String toXML()
    {
        return String.format("<triangle ax=\"%d\" ay=\"%d\" bx=\"%d\" by=\"%d\" cx=\"%d\" cy=\"%d\" />",ax,ay,bx,by,cx,cy);
    }

    @Override public String toString()
    {
        return String.format("Triangle(ax:%d,ay:%d,bx:%d,by:%d,cx:%d,cy:%d)",ax,ay,bx,by,cx,cy);
    }
}

File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06IO1/BridgePattern/BridgePattern/ShapeTriangle.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BridgePattern
{
    class ShapeTriangle : Shape
    {
        private static Brush bg = Brushes.Green;
        private float ax, ay, bx, by, cx, cy;
        public ShapeTriangle(string text)
        {
            ParseLambdas(text, v => ax = v, v => ay = v, v => bx = v, v => by = v, v => cx = v, v => cy = v);
        }
        public override void Draw(Graphics g)
        {
            g.FillPolygon
            (
                bg,
                new PointF[]
                {
                    new PointF(ax,ay),
                    new PointF(bx,by),
                    new PointF(cx,cy),
                }
            );
        }
        public override string ToXml()
        {
            return $"<triangle ax=\"{ax}\" ay=\"{ay}\" bx=\"{bx}\" by=\"{by}\" cx=\"{cx}\" cy=\"{cy}\" />";
        }
        public override string ToString()
        {
            return $"Triangle(ax:{ax}, ay:{ay}, bx:{bx}, by:{by}, cx:{cx}, cy:{cy})";
        }
    }
}